mcda/models/configuration.py

```diff
@@ -0,0 +1,218 @@
+
+        An array where the columns represent the indicators and the rows their values for different alternatives.  # noqa: E501
+
+        :return: The input_matrix of this Configuration.
+        :rtype: Dict[str, Dict[str, float]]
+        """
+        return self._input_matrix
+
+    @input_matrix.setter
+    def input_matrix(self, input_matrix: Dict[str, Dict[str, float]]):
+        """Sets the input_matrix of this Configuration.
+
+        An array where the columns represent the indicators and the rows their values for different alternatives.  # noqa: E501
+
+        :param input_matrix: The input_matrix of this Configuration.
+        :type input_matrix: Dict[str, Dict[str, float]]
+        """
+        if input_matrix is None:
+            raise ValueError("Invalid value for `input_matrix`, must not be `None`")  # noqa: E501
+
+        self._input_matrix = input_matrix
+
+    @property
+    def weights(self) -> List[float]:
+        """Gets the weights of this Configuration.
+
+        The weights associated to the indicators.  # noqa: E501
+
+        :return: The weights of this Configuration.
+        :rtype: List[float]
+        """
+        return self._weights
+
+    @weights.setter
+    def weights(self, weights: List[float]):
+        """Sets the weights of this Configuration.
+
+        The weights associated to the indicators.  # noqa: E501
+
+        :param weights: The weights of this Configuration.
+        :type weights: List[float]
+        """
+        if weights is None:
+            raise ValueError("Invalid value for `weights`, must not be `None`")  # noqa: E501
+
+        self._weights = weights
+
+    @property
+    def polarity(self) -> List[str]:
+        """Gets the polarity of this Configuration.
+
+        The polarity of an indicator indicates whether its rising value adds positively or negatively to the MCDA scores.  # noqa: E501
+
+        :return: The polarity of this Configuration.
+        :rtype: List[str]
+        """
+        return self._polarity
+
+    @polarity.setter
+    def polarity(self, polarity: List[str]):
+        """Sets the polarity of this Configuration.
+
+        The polarity of an indicator indicates whether its rising value adds positively or negatively to the MCDA scores.  # noqa: E501
+
+        :param polarity: The polarity of this Configuration.
+        :type polarity: List[str]
+        """
+        allowed_values = ["+", "-"]  # noqa: E501
+        if not set(polarity).issubset(set(allowed_values)):
+            raise ValueError(
+                "Invalid values for `polarity` [{0}], must be a subset of [{1}]"  # noqa: E501
+                .format(", ".join(map(str, set(polarity) - set(allowed_values))),  # noqa: E501
+                        ", ".join(map(str, allowed_values)))
+            )
+
+        self._polarity = polarity
+
+    @property
+    def sensitivity(self) -> ConfigurationSensitivity:
+        """Gets the sensitivity of this Configuration.
```

Comment:
Usually, these classes shouldn't be modified because they are generated by swagger and in future if we do any change to the API then all of these would be regenerated and it would be a complicated task to update them. 

I already feel that I have replicated `models` in two places. One here and the other in [ranking-service](https://github.com/futureforest-ai/ranking-service/tree/master/swagger_server/models). Somehow this must be avoided. For now I have left them as it is as it was a bit time consuming task. But if required then a follow up ticket must be created to further improve on refactoring.